waspc/data/Generator/templates/server/src/webSocket/initialization.ts

```diff
@@ -42,8 +43,12 @@ async function addUserToSocketDataIfAuthenticated(socket: Socket, next: (err?: E
   const sessionId = socket.handshake.auth.sessionId
   if (sessionId) {
     try {
-      const { user } =  await getSessionAndUserFromSessionId(sessionId)
-      socket.data = { ...socket.data, user }
+      const sessionAndUser =  await getSessionAndUserFromSessionId(sessionId)
+      const user = sessionAndUser ? makeAuthUserIfPossible(sessionAndUser.user) : null
```

Comment:
Can you do `makeAuthUserIfPossible(sessionAndUser?.user)` here?

I forgot how that questionmark thing works.

Reply:
Let me check it out and I'll update it if it makes sense 👍

Reply:
Given that the [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) short-circuits to `undefined` we should either need to update the `makeAuthUserIfPossible` definition to accept `undefined` or write it like this:

```ts
makeAuthUserIfPossible(sessionAndUser?.user) ?? null
```

We agreed offline that the ternary operator is more readable and we'll keep the existing logic.